packages/plugin-compiler/src/main/kotlin/io/realm/compiler/RealmModelSyntheticPropertiesGeneration.kt

```diff
@@ -346,6 +353,14 @@ class RealmModelSyntheticPropertiesGeneration(private val pluginContext: IrPlugi
                                         "Primary key ${property.name} is of type ${backingField.type.classifierOrFail.owner.symbol.descriptor.name} but must be of type ${validPrimaryKeyTypes.map { it.owner.symbol.descriptor.name }}",
                                     )
                                 }
+                                val indexableTypes = with(pluginContext.irBuiltIns) {
+                                    setOf(byteType, charType, shortType, intType, longType, stringType).map { it.classifierOrFail }
+                                }
+                                if (isIndexed && backingField.type.classifierOrFail !in indexableTypes) {
+                                    logError(
+                                        "Indexed key ${property.name} is of type ${backingField.type.classifierOrFail.owner.symbol.descriptor.name} but must be of type ${validPrimaryKeyTypes.map { it.owner.symbol.descriptor.name }}",
```

Comment:
The error message map the list of `validPrimaryKeyTypes`, we should probably have a `validIndexKeyTypes` instead